.github/workflows/test.yml

```diff
@@ -30,53 +36,178 @@ jobs:
+          run: |
+            bun install
+            bun test:unit
+
+  bun-linux-x64-musl:
+    name: Test bindings on linux-x64-musl
+    needs: zig-build-upload
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+      - name: Download artifacts
+        uses: actions/download-artifact@v4 # no need zig installation, download artifacts from previous job instead
+        with:
+          name: bindings-x86_64-linux-musl
+          path: zig-out/lib # download to the same folder of "zig build -Doptimize=ReleaseSafe"
+      - name: Setup and run tests using Bun
+        uses: addnab/docker-run-action@v3
+        with:
+          image: oven/bun:1.2.13-alpine
+          options: '-v ${{ github.workspace }}:/project -w /project/bun'
+          run: | # cannot run committeeIndices.test.ts and shuffle.test.ts due to musl is not available: @chainsafe/pubkey-index-map-linux-x64-musl
+            bun install
+            bun test test/unit/pubkeyIndexMap.test.ts
+
+  bun-linux-arm64-musl:
+    name: Test bindings on linux-arm64-musl
+    needs: zig-build-upload
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+      - name: Download artifacts
+        uses: actions/download-artifact@v4 # no need zig installation, download artifacts from previous job instead
+        with:
+          name: bindings-aarch64-linux-musl
+          path: zig-out/lib # download to the same folder of "zig build -Doptimize=ReleaseSafe"
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+        with:
+          platforms: arm64
+      - run: docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
+      - name: Setup and run tests using Bun
+        uses: addnab/docker-run-action@v3
+        with:
+          image: oven/bun:1.2.13-alpine
+          options: '--platform linux/arm64 -v ${{ github.workspace }}:/project -w /project/bun'
+          run: | # cannot run committeeIndices.test.ts and shuffle.test.ts due to musl is not available: @chainsafe/pubkey-index-map-linux-x64-musl
+            bun install
+            bun test test/unit/pubkeyIndexMap.test.ts
+
+  bun-macos:
+    name: Test bindings on macos and windows
+    needs: zig-build-upload
+    strategy:
+      matrix:
+        settings:
+          - os: macos-13
+            target: x86_64-macos-none
+          - os: macos-latest
+            target: aarch64-macos-none
+          - os: windows-latest
+            target: x86_64-windows-msvc
+    runs-on: ${{ matrix.settings.os }}
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+      - name: Download artifacts
+        uses: actions/download-artifact@v4 # no need zig installation, download artifacts from previous job instead
+        with:
+          name: bindings-${{ matrix.settings.target }}
```

Comment:
I suppose we could/should add another command to bun-ffi-z or tweak how it works to better support prebuilt libraries.

Current approach with `bun-ffi-z prepublish` is to rerun zig build for all supported targets.
It would be nice to support the devex of passing in a directory of subdirectories of prebuilt libraries (one directory per target) so that rebuild for all targets is unnecessary.

Then, the publish workflow would look like:
- download all artifacts to eg `artifacts` directory
- run modified bun-ffi-z prepublish to update all `targetPackages` (move each target library into the proper location for a publish)